src/main/webapp/resources/js/pages/admin/components/user-groups/AddUserToGroupButton.jsx

```diff
@@ -0,0 +1,179 @@
+  });
+
+  /*
+  Watch for changes to the debounced entered value for the user search.
+  Once it changes send a request for filtered users.
+   */
+  useEffect(() => {
+    if (debouncedQuery) {
+      getAvailableMembersFn(debouncedQuery).then((data) => setResults(data));
+    } else {
+      setResults([]);
+    }
+  }, [debouncedQuery, getAvailableMembersFn]);
+
+  /*
+  Watch for changes to the forms visibility, when it becomes visible
+  set keyboard focus onto the user name input.
+   */
+  useEffect(() => {
+    if (visible) {
+      setTimeout(() => userRef.current.focus(), 100);
+    }
+  }, [visible]);
+
+  const addMember = () => {
+    addMemberFn({ id: userId, projectRole })
+      .then((message) => {
+        addMemberSuccessFn();
+        notification.success({ message });
+        form.resetFields();
+        setVisible(false);
+      })
+      .catch((message) => notification.error({ message }));
+  };
+
+  /*
+  Before rendering format the results into Select.Option
+   */
+  const options = results.map((u) => (
+    <Option className="t-new-member" key={u.identifier}>
+      <Text style={{ marginRight: SPACE_XS }}>{u.label}</Text>
+      <Text type="secondary">{u.username}</Text>
+    </Option>
+  ));
+
+  const onCancel = () => {
+    form.resetFields();
+    setVisible(false);
+  };
+
+  return (
+    <>
+      <Button className="t-add-member-btn" onClick={() => setVisible(true)}>
+        {label}
+      </Button>
+      <Modal
+        className="t-add-member-modal"
+        visible={visible}
+        okButtonProps={{ disabled: typeof userId === "undefined" }}
+        onCancel={onCancel}
+        title={modalTitle}
+        onOk={addMember}
+        okText={i18n("AddMemberButton.modal.okText")}
+      >
+        <Form
+          layout="vertical"
+          form={form}
+          initialValues={{ role: projectRole }}
+        >
+          <Form.Item
+            label={i18n("AddMemberButton.modal.user-label")}
+            help={i18n("AddMemberButton.modal.user-help")}
+            name="user"
+          >
+            <Select
+              ref={userRef}
+              showSearch
+              notFoundContent={null}
+              onSearch={setQuery}
+              onChange={setUserId}
+              style={{ width: "100%" }}
+              value={userId}
+              filterOption={false}
+            >
+              {options}
+            </Select>
+          </Form.Item>
+          <Form.Item
+            label={i18n("AddMemberButton.modal.projectRole")}
+            name="projectRole"
+            initialValue={projectRole}
+          >
+            <Radio.Group
+              style={{ display: "flex" }}
+              onChange={(e) => setProjectRole(e.target.value)}
+            >
+              {projectRoles.map((role) => (
+                <Radio.Button key={role.value} value={role.value}>
+                  {role.label}
+                </Radio.Button>
+              ))}
+            </Radio.Group>
```

Comment:
Roles displayed here are not the group roles, they're the project roles.  This select doesn't function correctly.